### PR TITLE
Ensure that tokens are only sent with internal requests

### PR DESCRIPTION
**Pull request by dancancro**
- Please make sure the below checklist is followed for Pull Requests.

- [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [ ] Tests are added where necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by gmarziou** [comment]
Thanks Dan, maybe you could have described a little the context, I will as I was in touch with you about it.

Current implementation of AuthInterceptor adds the JWT token as the Authorization header to all HTTP requests, there are 2 problems with this approach:
- it can make some external APIs to reject requests
- it exposes user's tokens to other servers than the intended JHipster app which is a potential security issue

**Comment by gmarziou** [comment]
@jhipster/developers Dan's fix solves the problem in a simple way but I think we should consider reviewing our interceptors and our token storage solution, maybe by using a key that stores the domain name that emitted the token. Also I'm not sure how this would work with external auth providers like uaa, okta, keycloak, ...

**Comment by gmarziou** [comment]
I think it should also be considered in #6281

**Comment by Metavirulent** [comment]
In the context of UAA, we're about to get rid of auth.interceptor since access tokens will be kept in cookies and thus governed by the browser. Since cookies are bound to domains, they won't be sent  to other domains.

TL;DR: with PR #5812 merged, this will not have an impact on UAA-based microservices.

**Comment by ruddell** [comment]
Tested the AngularJS change and it works well.

The current behavior is that authorization token will not be sent unless your site is explicitly added to the remote site's Allowed Origins CORS headers.  Credentials will not be sent if the server has `*` for Allowed Origins ([MDN source](https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS#Access-Control-Allow-Origin)).

**Comment by xetys** [comment]
LGTM

**Comment by jdubois** [comment]
Thanks!!
